patches/minecraft/net/minecraft/client/gui/advancements/AdvancementsScreen.java.patch

```diff
@@ -45,28 +44,27 @@
        this.func_238696_c_(p_230430_1_, p_230430_2_, p_230430_3_, i, j);
        this.func_238695_a_(p_230430_1_, i, j);
        this.func_238697_d_(p_230430_1_, p_230430_2_, p_230430_3_, i, j);
-@@ -132,6 +145,7 @@
+@@ -132,6 +146,7 @@
           this.field_230706_i_.func_110434_K().func_110577_a(field_191945_g);
  
           for(AdvancementTabGui advancementtabgui : this.field_191947_i.values()) {
 +            if (advancementtabgui.getPage() == tabPage)
              advancementtabgui.func_238683_a_(p_238695_1_, p_238695_2_, p_238695_3_, advancementtabgui == this.field_191940_s);
           }
  
-@@ -139,6 +153,7 @@
+@@ -139,6 +154,7 @@
           RenderSystem.defaultBlendFunc();
  
           for(AdvancementTabGui advancementtabgui1 : this.field_191947_i.values()) {
 +            if (advancementtabgui1.getPage() == tabPage)
              advancementtabgui1.func_191796_a(p_238695_2_, p_238695_3_, this.field_230707_j_);
           }
  
-@@ -161,7 +176,7 @@
- 
+@@ -162,6 +178,7 @@
        if (this.field_191947_i.size() > 1) {
           for(AdvancementTabGui advancementtabgui : this.field_191947_i.values()) {
--            if (advancementtabgui.func_195627_a(p_238697_4_, p_238697_5_, (double)p_238697_2_, (double)p_238697_3_)) {
-+            if (advancementtabgui.getPage() == tabPage && advancementtabgui.func_195627_a(p_238697_4_, p_238697_5_, (double)p_238697_2_, (double)p_238697_3_)) {
+             if (advancementtabgui.func_195627_a(p_238697_4_, p_238697_5_, (double)p_238697_2_, (double)p_238697_3_)) {
++               if (advancementtabgui.getPage() == tabPage)
```

Comment:
Again, having a simple check at the start of the conditional makes it far more efficient in the negative case. So at the very least move the if statement up.